src/main/java/uk/gov/companieshouse/api/accounts/validation/CurrentPeriodValidator.java

```diff
@@ -26,13 +28,33 @@ public Errors validateCurrentPeriod(@Valid CurrentPeriod currentPeriod) {
         if (currentPeriod.getBalanceSheet() != null) {
 
             validateTotalFixedAssets(currentPeriod, errors);
+
+            validateTotalCurrentAssets(currentPeriod, errors);
         }
 
         return errors;
     }
-    
-    private void validateTotalFixedAssets(@Valid CurrentPeriod currentPeriod,
-        Errors errors) {
+
+    public void validateTotalCurrentAssets(CurrentPeriod currentPeriod, Errors errors) {
+
+        CurrentAssets currentAssets = currentPeriod.getBalanceSheet().getCurrentAssets();
+
+        Long stocks = currentAssets.getStocks();
```

Comment:
This could be shortened to this;:
```java
Long stocks = Optional.ofNullable(currentAssets.getStocks()).orElse(0L);
Long debtors = Optional.ofNullable(currentAssets. getDebtors()).orElse(0L);
Long cashAtBandAndInHand = Optional.ofNullable(currentAssets. getTotalCurrentAssets()).orElse(0L);

Long calculatedTotal = 0L;

calculatedTotal = stocks + debtors + cashAtBandAndInHand;

validateAggregateTotal(currentAssetsTotal, calculatedTotal, CURRENT_ASSETS_TOTAL_PATH, errors);
```